tomviz/python/tomviz/io/dm.py

```diff
@@ -0,0 +1,1332 @@
+"""
+A module to load data and meta data from DM3 and DM4 files into python
+as written by Gatan's Digital Micrograph program.
+
+Note
+----
+General users:
+    Use the simplified dm.dmReader() function to load the data and meta
+    data as a python dictionary.
+
+Advanced users and developers:
+    Access the file internals through the dm.fileDM() class.
+
+On Memory mode:
+  The fileDM support and "on memory" mode that pre-loads the file data in memory
+  and read operations during header parsing are performed against memory. This
+  can significantly improve performance when the file resides in a parallel file
+  system (PFS) because latency of seek operations PFSs is very high.
+
+This function is based on the Python code by Peter Ercius, NCEM
+- Berkeley National Lab.
+openNCEM - https://github.com/ercius/openNCEM
```

Comment:
OpenNCEM is GPL code looking at the repo, we could add it as an external package perhaps, but committing it like this is far from ideal.

Reply:
Would they relicense these files to 3-clause BSD?

Reply:
Sorry, I misread - I see from the text in the README it is dual licensed GPL or MIT. We can take MIT fine!

Reply:
@cryos huzzah! So is there anything you would like for me to do with these dm / ser file readers? Or are these scripts ok as is?

Reply:
Is this resolved, are we ok to merge here?

Reply:
The io portion of ncempy (openNCEM) was made as open as possible for this reason. You might want to use ncempy as a separate package though since you wont get any updates to readers if you copy/paste the code like this. DM3/4 and SER probably won't change anytime soon since those are quite mature file formats at this point.

Reply:
@jtschwar can you restrict your use to the io code of openNCEM? I think we are then ok.